service_restaurant/src/main/kotlin/org/thechance/service_restaurant/domain/utils/Validation.kt

```diff
@@ -108,7 +108,9 @@ class Validation : IValidation {
     }
 
     override fun isValidTime(time: String?): Boolean {
-        return time != null && time.matches(Regex("\\d{2}:\\d{2}"))
+        val pattern = "^([01]?[0-9]|2[0-3]):[0-5][0-9]$"
+        return Regex(pattern).matches(time!!)
```

Comment:
in this function the time parameter is nullable but there is not null assertion ( !! ) which will cause null pointer exception if the time passed as null